core/src/http/AuthHttpClient.js

```diff
@@ -207,11 +207,17 @@ export class AuthHttpClient {
      * @param {string} keyLevel - key level
      * @return {Object} response to the API call
      */
-    async getTransactions(accountId, offset, limit, keyLevel) {
+    async getTransactions(accountId, offset, limit, keyLevel, startDate, endDate) {
         this.useKeyLevel(keyLevel);
+        if (startDate === undefined) {
+            startDate = '';
+        }
+        if (endDate === undefined) {
+            endDate = '';
+        }
         const request = {
             method: 'get',
-            url: `/accounts/${accountId}/transactions?offset=${offset}&limit=${limit}`,
+            url: `/accounts/${accountId}/transactions?offset=${offset}&limit=${limit}&startDate=${startDate}&endDate=${endDate}`,
```

Comment:
You could get rid of the undefined evaluation above by doing `&startDate=${startDate || ''}&endDate=${endDate || ''}` unless you are considering `null` as a possible value